ksql-common/src/main/java/io/confluent/ksql/function/types/BooleanType.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.types;
+
+public final class BooleanType extends ObjectType {
+
```

Comment:
I think these classes are supposed to be Singletons, if so, would be nice to have a:

```
public static final BooleanType INSTANCE = new BooleanType();
```

And make the constructor private. Same applies for the other types.